lib/matplotlib/collections.py

```diff
@@ -106,9 +91,65 @@ def __init__(self,
                  **kwargs
                  ):
         """
-        Create a Collection
+        Parameters
+        ----------
+        edgecolors : default: :rc:`patch.edgecolor`
```

Comment:
this works for some types, but for things like "linestyles", that would make the thing:

`str or tuple or list of str or list of tuple`

which is 

1. less readable 
2. causes the line to overflow, and numpydocs [hasn't decided on how to deal with that yet](https://github.com/numpy/numpydoc/issues/87).

As a compromise, I've listed all the "scalar" types, along with a big notice at the top of the parameters section, just giving the user the heads up (since it's displayed right next to the class-level docstrings on the website, I think this should be enough).

Reply:
It's important to know which parameters support multiple values. The note at the top is a band-aid but I'm afraid that people will not see it.

Similar to #17446, but without the formality of defining classes, I intend to document such types in a central place and reference via `:ref:`linestyle` or a custom `:type:`linestyle`. That would then result in 
~~~
:ref:`linestyle` or list of :ref:`linestyle`
~~~
I just haven't come around doing so.

For now, would `[type] or list thereof` be ok?

Reply:
Oooh, I think "or list thereof" would fit actually.

The idea of making a `:type:\`linestyle\`` (and `color`, `capstyle`, etc.) is actually the center of my current GSOD proposal draft (that I should really get around to posting to discourse). 

So I think that we agree on the path forward. In the future we would want 

`:type:'linestyle' or list of :type:'linestyle'`, but for now I'll use the "or list thereof" as you suggest.